src/Utils.vala

```diff
@@ -427,45 +438,189 @@ public class FeedReader.Utils : GLib.Object {
 		return feedname.str;
 	}
 
-	public static bool downloadIconWithSession(Soup.Session session, string feed_id, string feed_url, string icon_path = GLib.Environment.get_user_data_dir() + "/feedreader/data/feed_icons/")
+	public static bool downloadFavIcon(string feed_id, string feed_url, string icon_path = GLib.Environment.get_user_data_dir() + "/feedreader/data/feed_icons/")
 	{
-		var path = GLib.File.new_for_path(icon_path);
-		try{path.make_directory_with_parents();}catch(GLib.Error e){}
-		string local_filename = icon_path + feed_id.replace("/", "_").replace(".", "_") + ".ico";
 
-		string url = feed_url;
-		if(feed_url.has_prefix("http://"))
-			url = url.replace("http://", "");
-		else if(feed_url.has_prefix("https://"))
-			url = url.replace("https://", "");
+		var uri = new Soup.URI(feed_url);
+		string hostname = uri.get_host();
+		int first = hostname.index_of_char('.', 0);
+		int second = hostname.index_of_char('.', first+1);
+		if(second != -1 && first != second)
+			hostname = hostname.substring(first+1);
+		string siteURL = uri.get_scheme() + "://" + hostname;
```

Comment:
Probably not worth the effort right now, but worth thinking about in the future I guess.